code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -495,38 +496,47 @@
  * scan_range - how far away from [src] will be scanned, if nothing is found directly adjacent.
  */
 /mob/living/simple_animal/bot/proc/scan(list/scan_types, old_target, scan_range = DEFAULT_SCAN_RANGE)
+	var/key = scan_types.Join(",")
+	var/list/scan_cache = GLOB.scan_typecaches[key]
+	if(!scan_cache)
+		scan_cache = typecacheof(scan_types)
+		GLOB.scan_typecaches[key] = scan_cache
 	var/turf/current_turf = get_turf(src)
 	if(!current_turf)
 		return
 	var/list/adjacent = current_turf.get_atmos_adjacent_turfs(1)
-	if(shuffle) //If we were on the same tile as another bot, let's randomize our choices so we dont both go the same way
+	if(shuffle_adjacent) //If we were on the same tile as another bot, let's randomize our choices so we dont both go the same way
 		adjacent = shuffle(adjacent)
-		shuffle = FALSE
-
-	for(var/turf/scan as anything in adjacent) //Let's see if there's something right next to us first!
-		if(check_bot(scan)) //Is there another bot there? Then let's just skip it
+		shuffle_adjacent = FALSE
+
+	var/list/turfs_to_walk = list()
+	for(var/turf/victim in view(scan_range, src))
+		turfs_to_walk += victim
+
+	turfs_to_walk = shuffle(turfs_to_walk - adjacent)
+	// Now we prepend adjacent since we want to run those first
+	turfs_to_walk = adjacent + turfs_to_walk
+
+	for(var/turf/scanned as anything in turfs_to_walk)
+		// Check bot is inlined here to save cpu time
+		//Is there another bot there? Then let's just skip it so we dont all atack on top of eachother.
+		var/bot_found = FALSE
+		for(var/mob/living/simple_animal/bot/buddy in scanned.contents)
```

Comment:
.contents has soft lookup overhead (Nadrew told me that)

Reply:
lookup overhead?

Reply:
> lookup overhead?

.contents isn't compile-optimized, so it still has to lookup the var. Byond!

Reply:
:(